### PR TITLE
Added usecase test for single tenancy details

### PR DESCRIPTION
**Pull request by LBHRShetty**
- Added usecase tests for Tenancy Details for Ref( TenancyDetailsByRef Execute)

### CODE REVIEW

File: LBHTenancyAPITest/Helpers/Fake.cs [hunk 1]
```diff
@@ -40,6 +42,50 @@ public static PaymentTransaction GeneratePaymentTransactionDetails()
...
+                       Frequency = random.Random.Hash(10),
+                       ClearBy = new DateTime(random.Random.Int(1900, 1999), random.Random.Int(1, 12), random.Random.Int(1, 28), 9, 30, 0),
+                       StartBalance = random.Finance.Amount(),
+                       Status = random.Random.Hash(10)
+                   }
+                }
+            };
+        }
+
+
```
Comment (StevenLeighton21) [inline LBHTenancyAPITest/Helpers/Fake.cs]: _whitespace ranger_

File: LBHTenancyAPITest/Test/UseCases/TenancyDetailsForRefTest.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        [Fact]
+        public void WhenGivenATenancyRefThatDoesntExist_ShouldReturnAnEmptyTenancyResponse()
+        {
+            var gateway = new StubTenanciesGateway();
+            var tenancyDetailsForRef = new TenancyDetailsForRef(gateway);
+
+            var response = tenancyDetailsForRef.Execute("");
+
+            Assert.IsType(typeof(TenancyDetailsForRef.TenancyResponse), response);
+            Assert.Equal(null,response.TenancyDetails.TenancyRef);
```
Comment (StevenLeighton21) [inline LBHTenancyAPITest/Test/UseCases/TenancyDetailsForRefTest.cs]: needs a space after the comma

**Comment by StevenLeighton21** [comment]
Description should describe what was changed and why

**Comment by StevenLeighton21** [comment]
I'd like to see the test generate more than one agreement and more than one action diary entry, just for sanity. Otherwise, LGTM!

File: LBHTenancyAPITest/Helpers/Fake.cs [hunk 3]
```diff
@@ -80,7 +88,16 @@ public static Tenancy GenerateTenancyDetails()
                        ClearBy = new DateTime(random.Random.Int(1900, 1999), random.Random.Int(1, 12), random.Random.Int(1, 28), 9, 30, 0),
                        StartBalance = random.Finance.Amount(),
                        Status = random.Random.Hash(10)
-                   }
+                   },
+                    new ArrearsAgreement()
```
Comment (StevenLeighton21) [inline LBHTenancyAPITest/Helpers/Fake.cs]: suggest this could now be in a helper function

File: LBHTenancyAPITest/Helpers/Fake.cs [hunk 4]
```diff
@@ -67,8 +67,16 @@ public static Tenancy GenerateTenancyDetails()
                         Date = new DateTime(random.Random.Int(1900, 1999), random.Random.Int(1, 12), random.Random.Int(1, 28), 9, 30, 0),
                         Comment = random.Random.Hash(100),
                         UniversalHousingUsername = random.Random.Hash(50)
-                    }
-
+                    },
+                    new ArrearsActionDiaryEntry()
```
Comment (StevenLeighton21) [inline LBHTenancyAPITest/Helpers/Fake.cs]: suggest this could now be in a helper function

**Comment by LBHRShetty** [comment]
Merging the branch now

**Pull request merged**